src/aap_eda/api/views/credential.py

```diff
@@ -127,6 +120,27 @@ def get_serializer_class(self):
     def get_response_serializer_class(self):
         return serializers.CredentialSerializer
 
+    @extend_schema(
+        description="List all credentials",
+        responses={
+            status.HTTP_200_OK: OpenApiResponse(
+                serializers.CredentialSerializer(many=True),
+                description="Return a list of credential.",
+            ),
+        },
+    )
+    def list(self, request):
+        credentials = models.Credential.objects.exclude(
+            credential_type=CredentialType.VAULT,
+            vault_identifier=EDA_SERVER_VAULT_LABEL,
```

Comment:
- It seems that we introduce a new "internal credential" type identified by `credential.credential_type == CredentialType.VAULT and credential.vault_identifier == EDA_SERVER_VAULT_LABEL` We already have a similar logic for destroy and I wonder if it would be more consistent and reusable through methods in the model. A method that returns a reusable queryset for non-internal credentials and a `is_internal` method that tell us if a credential is internal. 

- If we are not going to allow a deletion of a internal credential and we are not going to return them in the list view, does make sense to allow to fetch it by GET? It looks like the GET verb should have a similar logic to prevent returning an internal credential, don't?

- The previous two points make me wonder if it would not be better to have this internal credential in a separated table instead of taking care of excluding it in multiple places. @hsong-rh @benthomasson @mkanoor

Reply:
@Alex-Izquierdo what's your concern? I feel the current way is straightforward.

Reply:
@Alex-Izquierdo We can fix the GET but for the rest of the items can we punt now and do it later when the Gateway Credentials show up, we need the system credential for FanOut.

Reply:
approved.

Reply:
@mkanoor @Alex-Izquierdo The GET for internal credentials is hidden.